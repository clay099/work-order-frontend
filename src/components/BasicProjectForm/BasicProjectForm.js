import React from "react";
import FormAddressItems from "../FormAddressItems/FormAddressItems";
import { TextField, Grid } from "@material-ui/core";

/** BasicProjectForm
 * @param  {function} handleChange
 * @param  {object} formData
 *
 * Displays a Grid which renders:
 *    - TextField - for users to enter the new project description
 *    - FormAddressItems Component - renders the address items common in a number of forms
 *
 */
const BasicProjectForm = ({ handleChange, formData }) => {
	return (
		<>
			<Grid container spacing={2}>
				<Grid item xs={12}>
					<TextField
						name="description"
						variant="outlined"
						required
						fullWidth
						id="description"
						label="Project Description"
						autoFocus
						onChange={handleChange}
						value={formData.description}
					/>
				</Grid>
				<FormAddressItems
					formData={formData}
					handleChange={handleChange}
				/>
			</Grid>
		</>
	);
};

export default BasicProjectForm;
