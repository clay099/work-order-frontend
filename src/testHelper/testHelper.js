// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
