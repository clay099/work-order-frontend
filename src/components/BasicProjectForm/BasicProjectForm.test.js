import React from "react";
import BasicProjectForm from "./BasicProjectForm";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	handleChange: jest.fn(),
	formData: { description: "description" },
};

describe("BasicProjectForm Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<BasicProjectForm {...params} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<BasicProjectForm {...params} />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<BasicProjectForm {...params} />);
		const value = getByText("Project Description");
		expect(value).toBeInTheDocument();
	});
});
