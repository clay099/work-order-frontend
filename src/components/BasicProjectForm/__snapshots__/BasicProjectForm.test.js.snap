// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicProjectForm Component matches snapshot 1`] = `[Function]`;
